### PR TITLE
Added postfix configuration for MITx

### PR DESCRIPTION
**Pull request by bdero**
This PR contains the postfix configuration for MITx, including a root mail alias.

Resources used: 
- https://github.mit.edu/mitx-devops/mitx-ansible/blob/master/playbooks/roles/smtp/tasks/main.yml
- https://github.mit.edu/mitx-devops/mitx-ansible/blob/master/playbooks/roles/smtp/defaults/main.yml
- https://github.mit.edu/mitx-devops/mitx-ansible/blob/master/playbooks/roles/smtp/handlers/main.yml

@blarghmatey

### CODE REVIEW

File: salt/edx/smtp.sls [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+    - source: salt://edx/templates/postfix_main.cf.j2
+    - template: jinja
+
+forward_root_email:
+  alias.present:
+    - name: root
+    - target: {{ root_forward }}
+
+relay_postmap:
+  cmd.run:
```
Comment (blarghmatey) [inline salt/edx/smtp.sls]: If this only needs to get run when the postfix relay is updated then you should use `cmd.watch`

File: salt/edx/smtp.sls [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+    - name: /usr/sbin/postmap /etc/postfix/relay_passwd
+    - watch:
+      - file: configure_postfix_relay
+
+canonical_postmap:
+  cmd.run:
```
Comment (blarghmatey) [inline salt/edx/smtp.sls]: Same as above

File: salt/edx/smtp.sls [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+    - name: /usr/sbin/postmap /etc/postfix/canonical
+    - watch:
+      - file: create_canonical_file
+
+postfix_service:
+  service.running:
+    - enable: True
```
Comment (blarghmatey) [inline salt/edx/smtp.sls]: You need to specify a `- name:` parameter so that it knows which service to process.

File: salt/edx/smtp.sls [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+    - owner: root
+    - group: root
+    - mode: 600
+    - context:
+        root_from: {{ relay_host }}
+
+configure_postfix_relay:
+  file.managed:
+    - name: /etc/postfix.main.cf
+    - source: salt://edx/templates/postfix_main.cf.j2
```
Comment (blarghmatey) [inline salt/edx/smtp.sls]: It looks like this template didn't make it into this PR.

**Comment by bdero** [comment]
I addressed the comments here and added the missing template @blarghmatey . :)

**Comment by blarghmatey** [comment]
👍

**Comment by blarghmatey** [comment]
Just a reminder that this is good to go.

**Pull request merged**